troubadix/standalone_plugins/deprecate_vts.py

```diff
@@ -0,0 +1,264 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2024 Greenbone AG
+
+import os
+import re
+import sys
+from argparse import ArgumentParser, Namespace
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Iterable, Optional
+
+from troubadix.argparser import file_type
+from troubadix.helper.patterns import (
+    get_special_script_tag_pattern,
+    get_script_tag_pattern,
+    ScriptTag,
+    SpecialScriptTag,
+)
+
+
+class PathException(Exception):
+    pass
+
+
+@dataclass
+class DeprecatedFile:
+    name: str
+    full_path: Path
+    content: str
+
+
+# CHANGE AS NEEDED
+FILENAME_REGEX = re.compile(r"gb_rhsa_2021")
+
+KB_ITEMS = re.compile(r"set_kb_item\(.+\);")
+
+
+def update_summary(file: DeprecatedFile, oid_mapping_path: Path = None) -> str:
+    """Update the summary of the nasl script by adding the information
+    that the script has been deprecated, and if possible, the oid of
+    the new notus script replacing it.
+
+    Args:
+        file: DeprecatedFile object containing the content of the VT
+        oid_mapping_path (optional): The path to the file that contains a
+            mapping of old oids to new oids (see notus-generator transition
+            layer)
+
+    Returns:
+        The updated content of the file
+    """
+    if oid_mapping_path:
+        oid = match_oid(file.content, oid_mapping_path)
+        deprecate_text = f"This VT has been replaced by the new VT: {oid}. "
+    else:
+        deprecate_text = "This VT has been deprecated."
+
+    summary = get_summary(file.content)
+    if summary:
+        file.content = deprecate_text + summary
+    else:
+        print(f"No summary in: {file.name}")
+
+    return file.content
+
+
+def match_oid(content: str, oid_mapping_path: Path) -> str:
+    """Find the new Notus oid that has been mapped to the old
+    OID, so we can add this to the deprecation note.
+    """
+    pattern = get_special_script_tag_pattern(SpecialScriptTag.OID)
+    match = re.search(pattern, content)
+    old_oid = match.group(1)
+
+    # needs improvement
+    # pylint: disable=import-error, import-outside-toplevel
+    sys.path.append(oid_mapping_path)
+    from redhat import mapping
+
+    reverse_mapping = dict((v, k) for k, v in mapping.items())
+    new_oid = reverse_mapping.get(old_oid)
+    return new_oid
+
+
+def finalize_content(content: str) -> str:
+    """Update the content field of the nasl script by adding the
+    deprecated tag and removing the extra content."""
+    content_to_keep = content.split("exit(0);")[0]
+    return content_to_keep + (
+        "script_tag(name: 'deprecated', value: TRUE);"
```

Comment:
The usual / most common coding style of these is:

```
script_tag(name: 'deprecated', value: TRUE);
```

1. Double quotes instead of single ones
2. No space after the `:`